internal/util/tools/docs/docs.go

```diff
@@ -59,12 +60,12 @@ func main() {
 		}
 
 		fileLines := strings.Split(string(fileContents), "\n")
-		yamlSections := findYamlSections(fileLines)
+		jsonSections := findJsonSections(fileLines)
 
-		for _, yaml := range yamlSections {
-			_, _, report := config.Parse([]byte(strings.Join(yaml, "\n")))
-			reportStr := report.String()
-			if reportStr != "" {
+		for _, json := range jsonSections {
+			_, r, _ := config.Parse([]byte(strings.Join(json, "\n")))
+			reportStr := r.String()
+			if reportStr != "" && !isDeprecatedConfig(r) {
```

Comment:
Based on my reading of the `isDeprecatedConfig` function, this means that an error is not returned in the case of there being many items in the report and the first item is `ErrDeprecated`? So deprecated configs are effectively not checked?

Reply:
isDeprecatedConfig checks if the length of the report is 1 and returns false otherwise

Reply:
And I missed that too.